Sprintfinity3902/Controllers/KeyboardManager.cs

```diff
@@ -91,10 +89,10 @@ public class KeyboardManager : IUpdateable {
             RegisterCommand(new SetPlayerMoveLeftCommand((Player)player), Keys.A, Keys.Left);
             RegisterCommand(new SetPlayerMoveDownCommand((Player)player), Keys.S, Keys.Down);
             RegisterCommand(new SetPlayerMoveRightCommand((Player)player), Keys.D, Keys.Right);
-            
-            RegisterKeyUpCallback(
-                ((Player)player).CurrentState.Sprite.Animation.Stop,
-            Keys.W, Keys.A, Keys.S, Keys.D, Keys.Up, Keys.Down, Keys.Left, Keys.Right, Keys.E);
+
+            KeyboardManager.Instance.RegisterKeyUpCallback(() => {
+                ((Player)player).CurrentState.Sprite.Animation.Stop();
```

Comment:
This will NEED reviewed very soon. This can and likely will cause a failure state if player changes (i.e. goes into his damaged state). This casting has shown to be very dangerous and I have had to fix it in the past. especially since any key up will simply call to this command and live cast it.

The problem is that "player" swaps between regular and a decorated class edition. When its decorated, it CANNOT  be cast-ed to the type Player. Its a completely different version of ILink.

The solution I have found is to just save a cast-ed version of ILink in its regular state, and then call to that variable here in this command instead of trying to cast it live on key release.

Reply:
Scratch this, you shouldn't even need to cast it at all. the function initialize only takes type player, so no error should occur. I do think this same line is called in game1 at present and it does not need to be.